elastic/beats
# Add support for cloudfoundry metrics collection using consumer v1 API

## What does this PR do?

Continues with #19125                     to add support to Cloud Foundry consumer v1 API in Metricbeat too.

`version` can be used to select if `v1` or `v2` version should be used.

## Why is it important?

We already support the v2 API, but we saw that this API is not reliable enough (see elastic/beats#18202 and #18282).

## Checklist




## How to test this PR locally

- Run metricbeat with the cloudfoundry input and check that it collects events. Try these combinations of the `version` option:
  - No `version` option (what would use `v1`)
  - `version: v1`.
  - `version: v2`.
  - For other values of `version` it should fail to start.

## Related issues

- Continues with elastic/beats#19125
- Relates elastic/beats#16520
- Fixes elastic/beats#18282
